File: 136-single-number/single-number.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var singleNumber = function (nums) {
    const mp = new Map()
    for (let i = 0; i < nums.length; i++) {
        if (mp.has(nums[i])) {
            mp.set(nums[i], mp.get(nums[i]) + 1)
        } else {
            mp.set(nums[i], 1)
        }
    }
    for (const [key, value] of mp) {
        if (value === 1) {
            return key
        }
    }
};

File: 136-single-number/single-number.java
class Solution {
    public int singleNumber(int[] nums) {
        HashMap<Integer,Integer>  mp = new HashMap<>();

        for(int num: nums){
            mp.put(num, mp.getOrDefault(num, 0) + 1);
        }

        for(var num: mp.entrySet()){
            if(num.getValue() == 1){
                return num.getKey();
            }
        }

        return -1;
    }
}